src/Xamarin.Android.Build.Tasks/Tasks/Aapt2Link.cs

```diff
@@ -178,51 +178,58 @@ public async override System.Threading.Tasks.Task RunTaskAsync ()
 			
 			if (AdditionalResourceArchives != null) {
 				for (int i = AdditionalResourceArchives.Length - 1; i >= 0; i--) {
-					var flata = Path.Combine (WorkingDirectory, AdditionalResourceArchives [i].ItemSpec);
+					var flata = GetFullPath (AdditionalResourceArchives [i].ItemSpec);
 					if (Directory.Exists (flata)) {
 						foreach (var line in Directory.EnumerateFiles (flata, "*.flat", SearchOption.TopDirectoryOnly)) {
 							cmd.Add ("-R");
 							cmd.Add (GetFullPath (line));
 						}
 					} else if (File.Exists (flata)) {
 						cmd.Add ("-R");
-						cmd.Add (GetFullPath (flata));
+						cmd.Add (flata);
 					} else {
 						LogDebugMessage ("Archive does not exist: " + flata);
 					}
 				}
 			}
 
 			if (CompiledResourceFlatArchive != null) {
-				var flata = Path.Combine (WorkingDirectory, CompiledResourceFlatArchive.ItemSpec);
+				var flata = GetFullPath (CompiledResourceFlatArchive.ItemSpec);
 				if (Directory.Exists (flata)) {
 					foreach (var line in Directory.EnumerateFiles (flata, "*.flat", SearchOption.TopDirectoryOnly)) {
 						cmd.Add ("-R");
 						cmd.Add (GetFullPath (line));
 					}
 				} else if (File.Exists (flata)) {
 						cmd.Add ("-R");
-						cmd.Add (GetFullPath (flata));
+						cmd.Add (flata);
 				} else {
 					LogDebugMessage ("Archive does not exist: " + flata);
 				}
 			}
 
 			if (CompiledResourceFlatFiles != null) {
-				List<ITaskItem> appFiles = new List<ITaskItem> ();
+				var appFiles = new List<string> ();
 				for (int i = CompiledResourceFlatFiles.Length - 1; i >= 0; i--) {
 					var file = CompiledResourceFlatFiles [i];
-					if (!string.IsNullOrEmpty (file.GetMetadata ("ResourceDirectory")) && File.Exists (file.ItemSpec)) {
-						cmd.Add ("-R");
-						cmd.Add (GetFullPath (file.ItemSpec));
+					var fullPath = GetFullPath (file.ItemSpec);
+					if (!string.IsNullOrEmpty (file.GetMetadata ("ResourceDirectory"))) {
+						if (File.Exists (fullPath)) {
+							cmd.Add ("-R");
+							cmd.Add (fullPath);
+						} else {
+							LogDebugMessage ("File does not exist: " + fullPath);
+						}
 					} else {
-						appFiles.Add(file);
+						appFiles.Add(fullPath);
 					}
 				}
-				foreach (var file in appFiles) {
-					if (File.Exists (file.ItemSpec)) {
+				foreach (var fullPath in appFiles) {
+					if (File.Exists (fullPath)) {
```

Comment:
Perhaps we need to move the file exists check from here and  https://github.com/xamarin/xamarin-android/pull/5107/files#diff-a06c165fcdf5deb6c19ae2bf0f56dfedR217 to be outside these if/else statement and do a `Log` and `continue` if the file is missing...

might make the code a bit cleaner.